pkg/devfile/parser/configurables.go

```diff
@@ -0,0 +1,218 @@
+package parser
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/openshift/odo/pkg/application/labels"
+	"github.com/openshift/odo/pkg/config"
+	"github.com/openshift/odo/pkg/devfile/parser/data/common"
+	"github.com/openshift/odo/pkg/util"
+	"github.com/pkg/errors"
+)
+
+// This file contains all the parameters that can be change via odo config
+
+const (
+	Name              = "Name"
+	Ports             = "Ports"
+	Memory            = "Memory"
+	PortsDescription  = "Ports to be opened in all component containers"
+	MemoryDescription = "The Maximum memory all the component containers can consume"
+	NameDescription   = "The name of the component"
+)
+
+var (
+	supportedDevfileParameterDescriptions = map[string]string{
+		Name:   NameDescription,
+		Ports:  PortsDescription,
+		Memory: MemoryDescription,
+	}
+
+	lowerCaseDevfileParameters = util.GetLowerCaseParameters(GetDevfileSupportedParameters())
+)
+
+func (d DevfileObj) SetConfiguration(parameter string, value interface{}) error {
+
+	strValue, _ := value.(string)
```

Comment:
I think the error needs to be checked

Reply:
this was by design, think of it this way, if this fails you are the case statement where you are aware its not a strValue and you would use the `value` to change it to something u want.
Check the config.go - the same thing is happening there as well